app/helpers/application_helper.rb

```diff
@@ -29,11 +30,7 @@ def flash_class(level)
     end
   end
 
-  def peer_specialties_link(peer)
-    peer.specialties.map{|x| link_to(x.name, specialty_path(peer.order, x.slug))}.join(', ').html_safe
-  end 
-
-  def candidate_specialties_link(candidate)
-    candidate.specialties.map{|x| link_to(x.name, chambers_specialty_path(candidate.peerage_type, x.slug))}.join(', ').html_safe
+  def collection_link(collection)
+    collection.map{|x| link_to(*yield(x))}.join(', ').html_safe
```

Comment:
What does `*yield(x)` do? It's also not very clear to me what "collection" is or what its members are...